### ISSUE TITLE
author name string matching

### ISSUE THREAD

**Issue by sharifX** [CatalogueOfLife/checklistbank#999]
I am trying to understand how the author name string is used in the name matching tool. What should be the expected match type? 

I tested with a few incorrect author names (I added the string "blahblah" and my last name as test). I got the following responses: see the attached test result for details): 

1. Canonical: when I added my last name. 
2. Canonical: Also for typo and spelling error. I supplied "Agathis montana Lab"_. The correct name was _"Agathis montana de Laub."_ This was matched as canonical. 
3. None: When I added "blahblah"
4. Variant: Also for typo. _"Agathis montana Laub."_ match type was Variant. 

[Namematch_result_COL22.1_author_name_test.csv](https://github.com/CatalogueOfLife/checklistbank/files/8001999/Namematch_result_COL22.1_author_name_test.csv)

**Comment by mdoering** [comment]
I agree the match type can be confusing. At this stage it is probably best to ignore it as it is the match type to the names index only, not the target dataset, e.g. col. 
@thomasstjerne we should probably remove it until we have a true taxonomic matching. 

Canonical matches refer to matches ignoring the authorship and consider just the name alone. Maybe that term is too technical and sth like NAME_ONLY would be better?

Otherwise the names index matching tries to accept close variants of a name and author, bit not too distant changes. Your example given are expected and make sense to me.

**Comment by sharifX** [comment]
Thanks for the explanation. Yes, even though "canonical" might be a well known term for taxonomists it might not be clear in the string matching context here. Maybe a explanation somewhere like this would also be helpful: "The canonical name is the latinized elements of a scientific name".

**Comment by darpap** [comment]
So why in the forth example, with _Agathis montana Laub._ instead of _Agathis montana de Laub._ the match type has been "Variant" instead of "Canonical"? This is not very clear to me and so difficult to interpret.
I agree also that a clear definition of Canonical would be enough for non expert (like me), but a system that would also consider the author would be what we need (I'm responsible for an environmental european research infrastructure)

**Comment by mdoering** [comment]
The author matching is a bit lose, so `de Laub.` and `Laub.` are considered variants of the same authorship when found after the same canonical name. There is also an [authorship lookup file](https://github.com/CatalogueOfLife/backend/blob/master/api/src/main/resources/authorship/authormap.txt) which is used to assert `L.`, `Lin.`, `Linné` or `Linneaus` are the same for example.
